test/metabase/query_processor/pivot_test.clj

```diff
@@ -800,3 +800,33 @@
                 [nil 1 3]]
                (mt/rows
                 (qp.pivot/run-pivot-query query))))))))
+
+#_{:clj-kondo/ignore [:metabase/disallow-hardcoded-driver-names-in-tests]}
+(deftest pivot-table-from-native-mongo-query-test
+  (testing "#64124"
+    (mt/test-drivers #{:mongo}
+      (mt/dataset test-data
+        (let [query {:database (mt/id)
+                     :type     :query
+                     :query    {:source-table "card__1"
+                                :aggregation  [[:count]]
+                                :breakout     [[:field "PRODUCT_ID" {:base-type :type/Integer}]
+                                               [:field "QUANTITY" {:base-type :type/Integer}]]}}
+              pivot-settings {:pivot_table.column_split {:rows    ["PRODUCT_ID"]
+                                                         :columns ["QUANTITY"]
+                                                         :values  ["count"]}}]
+          (mt/with-metadata-provider
+            (lib.tu/mock-metadata-provider
+             (mt/metadata-provider)
+             {:cards [{:id 1
+                       :dataset-query {:type     :native
+                                       :native   {:collection "orders"
+                                                  :query      "[{\"$project\": {\"_id\": 0, \"PRODUCT_ID\": 1, \"QUANTITY\": 1}}]"}
+                                       :database (mt/id)}
+                       :database-id (mt/id)
+                       :result-metadata [{:name "PRODUCT_ID" :base_type :type/Integer}
+                                         {:name "QUANTITY" :base_type :type/Integer}]}]})
+            (let [result (qp/process-query (assoc query :visualization_settings pivot-settings))]
+              (is (= (count (:cols (:data result)))
+                     (count (first (:rows (:data result)))))
```

Comment:
Two things. If the query fails, this test will still pass, because `(count nil)` is `0`.

You can use `mt/cols` and `mt/rows` which will throw if the query has no rows/cols (because it failed) -- which will then trigger a test failure as we'd expect.

```suggestion
              (is (= (count (mt/cols result))
                     (count (first (mt/rows result))))
```

Second, I think it would be good to hardcode the expected column names to make sure that we don't somehow break the pivot queries by adding or removing a column. Something like

```clj
(is (= ["x" "y" "z"] (map :lib/desired-column-alias (mt/cols result))))
```

Reply:
Also, seeing how we're only looking at one row, can we maybe add a `:limit 1` or something to this query? So we don't burn up CI cycles running a query with thousands of rows where we only look at one? Not sure if this will break the pivot or not, but if we can do it we should